ui/src/middlewares/__tests__/logoutUser.test.jsx

```diff
@@ -0,0 +1,40 @@
+import middleware from '../logoutUser';
+import { userLogout } from '../../actions/user';
+
+jest.mock('../../actions/user');
+
+describe('LogoutUser middleware', () => {
+  let mirrorNext;
+  let dispatch;
+  let mockDispatch;
+
+  beforeEach(() => {
+    mirrorNext = jest.fn(value => value);
+    mockDispatch = jest.fn();
+    dispatch = middleware({ dispatch: mockDispatch })(mirrorNext);
+  });
+
+  afterEach(() => {
+    userLogout.mockClear();
+  });
+
+  it('dispatches logoutUser for SUBMIT_ERROR 401 and returns result of next(action)', () => {
+    const action = {
+      type: 'SUBMIT_ERROR',
+      payload: { status: 401 },
+    };
+    const result = dispatch(action);
+    expect(result).toBe(action);
+    expect(userLogout).toHaveBeenCalled();
+  });
+
+  it('only returns result of next(action) when not a SUBMIT_ERROR 401', () => {
+    const action = {
+      type: 'SOME_ERROR',
+      payload: { status: 500 },
```

Comment:
`status: 401` would be better, since it's not sure if that test is failing because it's not `SUBMIT_ERROR` or it is not `401`